c++/src/capnp/arena.h

```diff
@@ -91,11 +99,35 @@ class ReadLimiter {
 
 private:
   volatile uint64_t limit;
-  // Current limit, decremented each time catRead() is called.  Volatile because multiple threads
-  // could be trying to modify it at once.  (This is not real thread-safety, but good enough for
-  // the purpose of this class.  See class comment.)
+  // Current limit, decremented each time catRead() is called. We modify this variable using atomics
+  // with "relaxed" thread safety to make TSAN happy. See the class comment for why this is OK (
+  // previously we used a volatile variable).
 
   KJ_DISALLOW_COPY(ReadLimiter);
+
+  KJ_ALWAYS_INLINE(void setLimit(uint64_t newLimit)) {
+#if _MSC_VER && !defined(__clang__)
+#if _M_ARM
+    _InterlockedExchange64_nf(reinterpret_cast<volatile int64_t*>(&limit), newLimit);
+#else
+    _InterlockedExchange64(reinterpret_cast<volatile int64_t*>(&limit), newLimit);
```

Comment:
`_InterlockedExchange64` does an atomic exchange, not just a store, so is much more expensive than a store. (The function returns the old value of the target.)

Can we just leave it as a regular assignment on MSVC?